starfish/imagestack/imagestack.py

```diff
@@ -290,6 +290,131 @@ def xarray(self) -> xr.DataArray:
+        """Takes an xarray and key to index it. Indexes then adds back in lost dimensions"""
+        # store original dims
+        original_dims = data.dims
+        # index
+        data = data[key]
+        # find missing dims
+        missing_dims = set(original_dims) - set(data.dims)
+        # Add back in missing dims
+        data = data.expand_dims(tuple(missing_dims))
+        # Reorder to correct format
+        return data.transpose(*original_dims)
+
+    def _calc_new_coords(self, indexers: dict) -> xr.DataArray:
+        """Calculates the resulting coordinates array from indexing on each dimension in indexers
+
+        Parameters
+        ----------
+        indexers : a dictionary of dim:index where index is the value
+        or range to index the dimension
+
+        """
+        new_coords = self._coordinates.copy()
+        # index by R, CH, V
+        key = (indexers[Indices.ROUND.value], indexers[Indices.CH.value], indexers[Indices.Z.value])
+        new_coords = self._index_keep_dimensions(new_coords, key)
+        # check if X or Y dimension indexed, if so rescale
+        if self._needs_coords_resacling(indexers):
+            self._rescale_physical_coordinates(indexers, new_coords)
+        return new_coords
+
+    def _rescale_physical_coordinates(self, indexers: dict, new_coords: xr.DataArray) -> None:
+        """Iterates through coordinates array and rescales x, y physical coordinate values
+         based on how the x and y dimensions were indexed
+
+        Parameters
+        ----------
+        indexers : a dictionary of dim:index where index is the value
+        or range to index the dimension
+
+        new_coords: the coordinates xarray to modify
+
+        """
+        for _round in range(new_coords.sizes[Indices.ROUND]):
+            for ch in range(new_coords.sizes[Indices.CH]):
+                for z in range(new_coords.sizes[Indices.Z]):
+                    indices = {
+                        Indices.ROUND: _round,
+                        Indices.CH: ch,
+                        Indices.Z: z
+                    }
+                    xmin, xmax = self._rescale_physical_coordinate(
+                        Coordinates.X,
+                        indices,
+                        indexers[Indices.X.value])
+                    ymin, ymax = self._rescale_physical_coordinate(
+                        Coordinates.Y,
+                        indices,
+                        indexers[Indices.Y.value])
+
+                    new_coords[indices].loc[
+                        dict(physical_coordinate=[
+                            PhysicalCoordinateTypes.X_MIN,
+                            PhysicalCoordinateTypes.X_MAX,
+                            PhysicalCoordinateTypes.Y_MIN,
+                            PhysicalCoordinateTypes.Y_MAX]
+                        )] = [xmin, xmax, ymin, ymax]
+
+    def _rescale_physical_coordinate(self, coord: Coordinates, indices: Mapping[Indices, int],
```

Comment:
I don't think we're ever scaling in the sense that "scale" connotes a multiplication by a constant value -- we're "sub-setting" or "slicing", or something like that, right?

Reply:
Yea good point, well we are doing some scaling operations to determine the corresponding physical coordinate of a new pixel one...but there's probably a better name for that. Maybe like calculate_new_pyhsical_coordinate? ...I'm open to suggestions haha